File: algorithms/0792.binary_search/binary_search.cpp
class Solution {
public:
   int search(const vector<int>& nums, int target) {
       int lo = 0, hi = nums.size() - 1;
        while (lo <= hi) {
            const int mi = lo + (hi - lo) / 2;
            if (nums[mi] == target) {
                return mi;
            } else if (nums[mi] < target) {
                lo = mi + 1;
            } else {
                hi = mi - 1;
            }
        }
        return -1;
    }
};

File: algorithms/0792.binary_search/binary_search.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let (mut lo, mut hi) = (0i32, nums.len() as i32);
        while lo < hi {
            let mi = (lo + (hi - lo) / 2) as usize;
            if nums[mi] == target {
                return mi as i32;
            } else if nums[mi] < target {
                lo = mi as i32 + 1;
            } else {
                hi = mi  as i32;
            }
        }
        -1
    }
}